### PR TITLE
Feature | training-getweather-component

### PR DESCRIPTION
**Pull request by akashsen-aurx**

### CODE REVIEW

File: core/src/main/java/com/aurx/core/models/DisableArticleSchemaModel.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import org.apache.sling.models.annotations.Model;
+import org.apache.sling.models.annotations.injectorspecific.ScriptVariable;
+import org.apache.sling.models.annotations.injectorspecific.SlingObject;
+
+import javax.annotation.PostConstruct;
+
+@Model(adaptables = {Resource.class, SlingHttpServletRequest.class}, defaultInjectionStrategy = DefaultInjectionStrategy.OPTIONAL)
+public class DisableArticleSchemaModel {
+
+    /*@SlingObject
```
Comment (abhinav-aggarwal-aem) [inline core/src/main/java/com/aurx/core/models/DisableArticleSchemaModel.java:16]: why this commented line

File: core/src/main/java/com/aurx/core/services/config/GetWeatherConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package com.aurx.core.services.config;
+
+import org.osgi.service.metatype.annotations.AttributeDefinition;
+import org.osgi.service.metatype.annotations.ObjectClassDefinition;
+
+@ObjectClassDefinition(name = "GetWeather Endpoint Configuration")
+public @interface GetWeatherConfiguration {
+  @AttributeDefinition(name="Weather Report URL")
```
Comment (abhinav-aggarwal-aem) [inline core/src/main/java/com/aurx/core/services/config/GetWeatherConfiguration.java:8]: please give label and description as well

File: core/src/main/java/com/aurx/core/services/impl/GetWeatherServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      try {
+        URL url = new URL(baseURL);
+        logger.debug("get Url : {}", url);
+        HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+        connection.setRequestProperty("accept", "application/json");
+        InputStream responseStream = connection.getInputStream();
+        logger.debug("get responceStream:{}", responseStream);
+        response = IOUtils.toString(responseStream, StandardCharsets.UTF_8);
+        logger.debug("get responce : {}", response);
+      } catch (Exception ee) {
```
Comment (abhinav-aggarwal-aem) [inline core/src/main/java/com/aurx/core/services/impl/GetWeatherServiceImpl.java:48]: why catching base Exception class

File: core/src/main/java/com/aurx/core/servlets/GetWeatherServlet.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+@Component(service = Servlet.class, immediate = true, property = {
+    "sling.servlet.methods=POST",
+    "sling.servlet.resourceTypes=akash-project/components/getweather",
+    "sling.servlet.selectors=getWeather",
+    "sling.servlet.extensions=json"
+})
+public class GetWeatherServlet extends SlingAllMethodsServlet {
+
+  Logger logger = LoggerFactory.getLogger(GetWeatherServlet.class);
```
Comment (abhinav-aggarwal-aem) [inline core/src/main/java/com/aurx/core/servlets/GetWeatherServlet.java:30]: why is it not private and static and final

File: core/src/main/java/com/aurx/core/servlets/GetWeatherServlet.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+  private GetWeatherService getWeatherService;
+
+  @Override
+  protected void doPost(SlingHttpServletRequest request, SlingHttpServletResponse response)
+      throws ServletException, IOException {
+    String userIp = request.getParameter("userIp");
+    JsonObject jsonObject = getWeatherService.fetchLatLon(userIp);
+    if (jsonObject != null) {
+      String lat = jsonObject.get("lat").getAsString();
+      String lon = jsonObject.get("lon").getAsString();
```
Comment (abhinav-aggarwal-aem) [inline core/src/main/java/com/aurx/core/servlets/GetWeatherServlet.java:41]: what is lon?

File: core/src/main/java/com/aurx/core/servlets/GetWeatherServlet.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+  @Reference
+  private GetWeatherService getWeatherService;
+
+  @Override
+  protected void doPost(SlingHttpServletRequest request, SlingHttpServletResponse response)
+      throws ServletException, IOException {
+    String userIp = request.getParameter("userIp");
+    JsonObject jsonObject = getWeatherService.fetchLatLon(userIp);
+    if (jsonObject != null) {
+      String lat = jsonObject.get("lat").getAsString();
```
Comment (abhinav-aggarwal-aem) [inline core/src/main/java/com/aurx/core/servlets/GetWeatherServlet.java:40]: please use correct variable names

File: core/src/main/java/com/aurx/core/servlets/GetWeatherServlet.java [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+public class GetWeatherServlet extends SlingAllMethodsServlet {
+
+  Logger logger = LoggerFactory.getLogger(GetWeatherServlet.class);
+  @Reference
+  private GetWeatherService getWeatherService;
+
+  @Override
+  protected void doPost(SlingHttpServletRequest request, SlingHttpServletResponse response)
+      throws ServletException, IOException {
+    String userIp = request.getParameter("userIp");
```
Comment (abhinav-aggarwal-aem) [inline core/src/main/java/com/aurx/core/servlets/GetWeatherServlet.java:37]: where are loggers

File: core/src/main/java/com/aurx/core/models/DisableArticleSchemaModel.java (hunk 1 above)
Comment (akashsen-aurx) [inline core/src/main/java/com/aurx/core/models/DisableArticleSchemaModel.java:16]: remove comments

File: core/src/main/java/com/aurx/core/servlets/GetWeatherServlet.java [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+    JsonObject jsonObject = getWeatherService.fetchLatLon(userIp);
+    if (jsonObject != null) {
+      String lat = jsonObject.get("lat").getAsString();
+      String lon = jsonObject.get("lon").getAsString();
+      String units = "metric";
+      JsonObject weatherReport = getWeatherService.getWeatherReport(lat, lon, units);
+      List<WeatherInfo> weatherInfoList = new ArrayList<>();
+      if (weatherReport != null) {
+        JsonArray responseArray = weatherReport.getAsJsonArray("daily");
```
Comment (abhinav-aggarwal-aem) [inline core/src/main/java/com/aurx/core/servlets/GetWeatherServlet.java:46]: what is daily is not present in response

File: core/src/main/java/com/aurx/core/servlets/GetWeatherServlet.java [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+        for (JsonElement element : responseArray) {
+          int sunrise = element.getAsJsonObject().get("sunrise").getAsInt();
```
Comment (abhinav-aggarwal-aem) [inline core/src/main/java/com/aurx/core/servlets/GetWeatherServlet.java:48]: whta if sunrise is not present in response.

File: ui.apps/src/main/content/jcr_root/apps/akash-project/components/getweather/clientlibs/getweatherlib/js/main.js [hunk 10]
```diff
@@ -0,0 +1,46 @@
+window.onload = function () {
+
+    var path = document.querySelectorAll(".currentNodePath")[0].value;
+    var numberOfDay = document.querySelectorAll(".numberOfDay")[0].value;
+
+
+
+
+    var url = "http://localhost:4502" + path + ".getWeather.json";
+    console.log("akash sen");
```
Comment (abhinav-aggarwal-aem) [inline ui.apps/src/main/content/jcr_root/apps/akash-project/components/getweather/clientlibs/getweatherlib/js/main.js:10]: please remove all console logs

File: ui.apps/src/main/content/jcr_root/apps/akash-project/components/productDetail/clientlibs/productstyle/js/main.js [hunk 11]
```diff
@@ -0,0 +1,27 @@
+
+console.log("hiiii");
+document.addEventListener("DOMContentLoaded",function(){
+    console.log("loaded");
```
Comment (abhinav-aggarwal-aem) [inline ui.apps/src/main/content/jcr_root/apps/akash-project/components/productDetail/clientlibs/productstyle/js/main.js:4]: please remove console logs.

**Review by abhinav-aggarwal-aem — CHANGES_REQUESTED** [review]

**Review by abhinav-aggarwal-aem — APPROVED** [review]

**Pull request merged**